appnav/src/main/kotlin/io/element/android/appnav/RootFlowNode.kt

```diff
@@ -231,17 +233,23 @@ class RootFlowNode @AssistedInject constructor(
         Timber.d("Navigating to $deeplinkData")
         attachSession(deeplinkData.sessionId)
             .apply {
-                val roomId = deeplinkData.roomId
-                if (roomId == null) {
-                    // In case room is not provided, ensure the app navigate back to the room list
-                    attachRoot()
-                } else {
-                    attachRoom(roomId)
-                    // TODO .attachThread(deeplinkData.threadId)
+                when (deeplinkData) {
+                    is DeeplinkData.Root -> attachRoot()
+                    is DeeplinkData.Room -> navigateToRoom(deeplinkData)
+                    is DeeplinkData.InviteList -> navigateToInviteList(deeplinkData)
                 }
             }
     }
 
+    private suspend fun LoggedInFlowNode.navigateToRoom(deeplinkData: DeeplinkData.Room) {
+        backstack.push(LoggedInFlowNode.NavTarget.Room(deeplinkData.roomId))
+    }
+
+    private suspend fun LoggedInFlowNode.navigateToInviteList(deeplinkData: DeeplinkData.InviteList) {
+        notificationDrawerManager.clearMembershipNotificationForSession(deeplinkData.sessionId)
+        backstack.push(LoggedInFlowNode.NavTarget.InviteList)
+    }
```

Comment:
Done in https://github.com/vector-im/element-x-android/pull/735/commits/d8f31035e852ab0987036b3138cc703e5c7f5c43

Reply:
Sorry my first comment was not complete enough. I would prefer those 2 fun to be moved to the class `LoggedInFlowNode`. It will be clearer which backstack is used, and I do not really like that the `NavTarget` and the `backstack` are used outside of the class they are defined.